modules/ROOT/pages/access-source-message-attributes.adoc

```diff
@@ -0,0 +1,250 @@
+** subject
+** fileSize 
+* customProtocolHeaders
++
+For security reasons, only the headers with the prefix 'x-apm-' are available in the 'b2bMessageContext' variable.
+
+For example:
+
+[source,json]
+----
+{
+  "b2bMessageContext": {
+    "sourceProtocol": "AS2",
+    "transmissionId": "06bc20c8-475e-4a6f-b010-9b7933b2bfac",
+    "sourceEndpointAttributes": {
+        "as2MessageId": "<2023-04-12T19:37:42.273-46b95696-83f4-4e65-9222-61c40a854200@partnera_mythical-as2>",
+        "as2From": "partnera",
+        "as2To": "mythical-as2",
+        "fileName": "EDI_850_20230401_1245678.edi",
+        "subject": "EDI Message via Mule Generic Service",
+        "fileSize": 3887
+    },
+    "customProtocolHeaders": {
+      "x-apm-order-type": "drop-ship",
+      "x-apm-tracking-id": "9d26d4ce-59ea-4448-8b51-ccc975d4efec"
+    },
+    "messageId": "9bf446f8-4a02-475f-afa3-c8772a340154"
+  }
+}
+----
+
+== FTP and SFTP Source Endpoint
+ 
+Messages received via FTP or SFTP contain the following fields in the `b2bMessageContext` variable:
+
+* sourceProtocol
+* transmissionId
+* sourceEndpointAttributes:
+** fileName
+** filePath
+** fileSize
+** fileTimeStamp
+** partnerReferenceId
++
+This attribute is available only for Receive from Partner endpoints that have the partner reference ID configured.
++
+* messageId
+
+For example:
+
+[source,json]
+----
+{
+  "b2bMessageContext": {
+    "sourceProtocol": "SFTP",
+    "transmissionId": "3ef5654a-5cda-41b1-8909-59a28c8b1537",
+    "sourceEndpointAttributes": {
+      "fileName": "RandomMessageType_PO_20230331_1559_003.xml",
+      "filePath": "/partners/pacific-corp/in-orders",
+      "fileSize": 4889,
+      "fileTimeStamp": "2023-04-05T15:16:29",
+      "partnerReferenceId": "PACIFICCORP"
+    },
+    "messageId": "cdaf6c12-bb3b-4978-9ac6-e8ace053f7c5"
+  }
+}
+----
+
+== HTTP and HTTPS Source Endpoint
+
+Messages received via HTTP or HTTPS contain the following fields in the `b2bMessageContext` variable:
+
+* sourceProtocol
+* transmissionId
+* sourceEndpointAttributes:
+** path
+** fileSize
+* customProtocolHeaders
++
+For security reasons, only the headers with the prefix 'x-apm-' are available in the 'b2bMessageContext' variable.
+* messageId   
+
+For example:
+
+[source,json]
+----
+{ 
+  "b2bMessageContext": {
+    "sourceProtocol": "HTTPS",
+    "transmissionId": "a7a800ad-4117-40c1-98a0-79892c982fd4",
+    "sourceEndpointAttributes": {
+    "path": "/",
+    "fileSize": 866
+    },
+    "customProtocolHeaders": {
+      "x-apm-order-number": "ORD12345-A",
+      "x-apm-order-type": "drop-ship"
+    },  
+    "messageId": "cdaf6c12-bb3b-4978-9ac6-e8ace053f7c5"
+  }
+}
+----
+
+[[implementation-examples]]
+== Use Case Implementation Examples
+
+The following examples show how to implement some common use cases for source message attributes.
+
+This example shows how to map the `messageId` and `fileName` attributes in the translated data payload within a message flow's translation map:
```

Comment:
One thought to streamline and improve scannability is to remove the "This example shows how to" intro from each and make these headings or bullets.